compiler/rustc_resolve/src/late.rs

```diff
@@ -580,6 +580,14 @@ impl MaybeExported<'_> {
     }
 }
 
+/// Used for recording UnnecessaryQualification.
+#[derive(Debug)]
+pub(crate) struct UnnecessaryQualification<'a> {
+    pub binding: LexicalScopeBinding<'a>,
+    pub finalize: Finalize,
```

Comment:
Only `node_id` and `path_span` are used, can keep just them instead of full `Finalize`.